src/validation.cpp

```diff
@@ -5051,6 +5172,34 @@ void ChainstateManager::Unload()
     m_best_invalid = nullptr;
 }
 
+void ChainstateManager::cleanup()
```

Comment:
In commit "validation: add ChainMan logic for completing UTXO snapshot validation" (328aa05c28fe9e5fff5d90e9b23bbbb5f6da1846)

I think I'd NACK the approach for trying to rename and delete things on shutdown. I don't see why it's not possible to rename and delete the snapshot when it is validated. I understand that maybe you want the pointers to be stable so can't delete the CChainstate object, but that doesn't seem like a reason why you can't close/destroy databases and rename directories.

If I'm missing something and it really is not possible to rename the chainstate directories while the node is running, the best time to rename them would be startup time, not shutdown time. Shutdown is the time to stop work, not begin it, and it's especially precarious if bitcoin is running on a constantly rebooting windows machine or as a systemd service, where the system will may nicely for a shutdown but quickly run out of patience and kill the process.

Reply:
Good points, willfix.

Reply:
I'm now doing the "cleanup" (i.e. deletion of background chainstate data) on startup, and not shutdown. I'd like to experiment with doing deletion inline in a follow-up PR.